redfish-core/lib/log_services.hpp

```diff
@@ -3346,4 +3415,931 @@ inline void requestRoutesPostCodesEntry(App& app)
+        {
+            BMCWEB_LOG_ERROR
+                << "Failed to get parent Redfish URI DBus interface for the "
+                   "identified Redfish URI: "
+                << redfishUri
+                << " of the given DBus object path: " << dbusObjPath.str;
+            return;
+        }
+
+        crow::connections::systemBus->async_method_call(
+            [asyncResp, entryJsonIdx, dbusObjPath, uriIdPos, uriIdPattern,
+             parentRedfishUri,
+             redfishUri](const boost::system::error_code ec,
+                         const std::vector<std::string>& objects) mutable {
+                if (ec || objects.empty())
+                {
+                    BMCWEB_LOG_ERROR << "Failed to get parent DBus path for "
+                                        "identified parent Redfish URI: "
+                                     << parentRedfishUri
+                                     << " of the given DBus object path: "
+                                     << dbusObjPath.str;
+                    return;
+                }
+
+                std::string uriIdObj;
+                for (auto& object : objects)
+                {
+                    if (dbusObjPath.str.find(object) != std::string::npos)
+                    {
+                        uriIdObj = object;
+                        break;
+                    }
+                }
+
+                if (uriIdObj.empty())
+                {
+                    BMCWEB_LOG_ERROR << "Failed to get parent DBus path for "
+                                        "identified parent Redfish URI: "
+                                     << parentRedfishUri
+                                     << " of the given DBus object path: "
+                                     << dbusObjPath.str;
+                    return;
+                }
+
+                redfishUri.replace(
+                    uriIdPos, uriIdPattern.length(),
+                    getIsolatedHwItemId(
+                        sdbusplus::message::object_path(uriIdObj)));
+
+                if (redfishUri.find(uriIdPattern) == std::string::npos)
+                {
+                    nlohmann::json& entryJson =
+                        (entryJsonIdx > 0
+                             ? asyncResp->res
+                                   .jsonValue["Members"][entryJsonIdx - 1]
+                             : asyncResp->res.jsonValue);
+
+                    entryJson["Links"]["OriginOfCondition"] = {
+                        {"@odata.id", redfishUri}};
+                }
+                else
+                {
+                    fillRedfishUriWithIds(asyncResp, entryJsonIdx, dbusObjPath,
+                                          redfishUri);
+                }
+            },
+            objMapperSerName, objMapperObjPath, objMapperIface,
+            "GetSubTreePaths", "/xyz/openbmc_project/inventory", 0,
+            std::get<0>(*parentRedfishUriIt));
```

Comment:
common error 5

Reply:
Agree on the common error 5 but, here I am using for tuple, not for the vector. 
I think tuple won't throw an exception. FYR https://en.cppreference.com/w/cpp/utility/tuple/get - Even that bmcweb common error did not mention for the `tuple` - I am missing anything on this?